src/opm/parser/eclipse/EclipseState/Schedule/MSW/Segment.cpp

```diff
@@ -209,6 +209,9 @@ namespace {
         return m_depth;
     }
 
+    double Segment::perfLength() const {
+        return *this->m_perf_length;
```

Comment:
is this `*` intentional?

Reply:
Yes - that is intentional; the `m_perf_length` member is a `std::optional<double>` as a way to safeguard against accessing an uninitialized value. A bit more verbose could be:
```C++
if (this->m_perf_length)
     return *this->m_perf_length;
throw std::logic_error("Tryng to access an uninitialized perforation length variable");
```